.azure-pipelines/azure-pipelines-backend.yml

```diff
@@ -3,43 +3,95 @@ trigger:
     include:
       - master
   paths:
-    exclude: 
+    exclude:
       - /.azure-pipelines/azure-pipelines-frontend.yml
       - /packages/client/*
       - /packages/client-e2e/*
 
 pool:
   vmImage: 'ubuntu-latest'
 
-steps: 
+steps:
   - task: NodeTool@0
     inputs:
       versionSpec: '14.6.0'
     displayName: 'Install Node.js 14.6.0'
 
+  # This is included in package.json, but only under devDependencies, so this (production!) build won't see it.
+  # As a workaround, install it globally
+  - script: yarn global add wasm-pack
+    displayName: 'Install wasm-pack'
+
+  - task: Cache@2
+    inputs:
+      # Cache the tarred node_modules.xz file.
+      # Note that the cache key is the hash of of yarn.lock. If the lockfile changes, the cache gets invalidated.
+      key: 'v2 | yarn | "$(Agent.OS)" | $(Build.SourcesDirectory)/yarn.lock'
+      path: '$(Build.SourcesDirectory)/tarred_node_modules'
+      cacheHitVar: 'NodeCacheRestored'
+    displayName: 'Cache node_modules'
+
+  # Untar the cached node_modules folder if we had a cache hit
+  - script: tar -xf $(Build.SourcesDirectory)/tarred_node_modules/node_modules.tar.xz --directory=$(Build.SourcesDirectory) --verbose
+    condition: eq(variables['NodeCacheRestored'], 'true')
+    displayName: 'Untar node_modules.xz'
```

Comment:
...and if it's in the cache, we just pull it out and untar it. Which is much faster than getting the folder out of cache, then tarring it for publishing at the end!